examples/meta/generator/targets/cpp.json

```diff
@@ -102,7 +102,8 @@
             "get_int_string_list": "$object->get<SGStringList<int32_t>>($arguments)",
             "get_char_string_list": "$object->get<SGStringList<char>>($arguments)",
             "get_word_string_list": "$object->get<SGStringList<uint16_t>>($arguments)",
-            "get_option": "$object->get<std::string>($arguments)"
+            "get_option": "$object->get<std::string>($arguments)",
+            "get_CV_storage": "$object->get<CrossValidationStorage*>($arguments)"
```

Comment:
This type should be hidden from swig / interfaces or?

Reply:
It is needed if we want to make the user able to extract it from the parameter map with `get`.

Reply:
But why does the specialised class need be exposed and not the base class? Get is sgobject

Reply:
My point is: the CV storage type is not needed in the interfaces, just a getter for the base type and then the user can use ‘get’

Reply:
That's because `CrossValidationStorage` has some methods to store/get objects (e.g. `get_fold()`) which are not present in SGObject. I guess that I could remove all those methods and use directly the tag framework. We could then remove `get_CV_storage`.

Reply:
Yes we actually need to do this. Tag framework is the way to go. Reason is all these things will be plugins at some point and we also want to make swig api stable

Reply:
Yes we actually need to do this. Tag framework is the way to go. Reason is all these things will be plugins at some point and we also want to make swig api stable

Reply:
updates on this?

Reply:
I am trying to see if I can use just the standard `get()` and remove this guy. Currently I am trying to add the `CrossValidationStorage` to this method [here](https://github.com/shogun-toolbox/shogun/blob/develop/src/shogun/base/SGObject.h#L1100). After that it should work.